### PR TITLE
Use correct filename(s) for binary warning

### LINKED ISSUE #411: Validator misflags png as binary when another binary file exists
https://addons.mozilla.org/en-US/developers/addon/flash-debugger/file/421089/validation

```
Flagged file extensions found.

Warning: Files whose names end with flagged extensions have been found in the add-on.

The extension of these files are flagged because they usually identify binary components. 
Please see http://addons.mozilla.org/developers/docs/policies/reviews#section-binary for 
more information on the binary content review process.

data/flashfirebug.swf
    data/html/images/themes/default/inspector/UIComponents/VolumeBar.png
```

The png file is not a binary file, nor should it be flagged.

### PR DESCRIPTION
**Pull request by wagnerand**
Fixes #411

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
I would love to see a minimal test verifying that the output is indeed what we expect since now `filename` is a comma separated list and cannot be resolved to a particular filename anymore.

Isn't the problem in #411 a different one? I mean, the error reported there is also referencing the actual description which uses `flagged_files` so somehow the PNG ends up there.

I don't see exactly why that should be the case (the code is relatively simple, if extension in blacklisted_extensions... and there is no png in there) but maybe a test reveals a bit more details.

**Comment by wagnerand** [comment]
The problem is that in https://github.com/mozilla/amo-validator/blob/master/validator/testcases/packagelayout.py#L65 we loop over all files to identify the ones we should flag. However handling of flagged files is outside the loop (https://github.com/mozilla/amo-validator/blob/master/validator/testcases/packagelayout.py#L94) but we use the loop variable `name` in https://github.com/mozilla/amo-validator/blob/master/validator/testcases/packagelayout.py#L125 for the filename to flag, which happens to be the last checked file from the loop.

I considered moving the warning into the loop, but it's generally more useful if we have one warning and just list _n_ occurrences instead of having _n_ warnings with one occurrence each.

**Comment by EnTeQuAk** [comment]
fair enough, let's try this and see if there's more hiding.

But can you add a very simple test at least somewhere in `tests/test_packagelayout.py`? There are already a few and I believe it should be reasonably easy to add a simple one to verify it's working as expected.

**Comment by wagnerand** [comment]
@EnTeQuAk I adapted an existing test. Mind having another look?

File: tests/test_packagelayout.py [hunk 1]
```diff
@@ -17,6 +17,8 @@ def test_blacklisted_files():
                    packagelayout.test_blacklisted_files,
                    True)
     assert err.metadata['contains_binary_extension']
+    any(warning['id'][1] == 'test_blacklisted_files'
```
Comment (EnTeQuAk) [inline tests/test_packagelayout.py]: guess there's an `assert` missing?

Comment (wagnerand) [inline tests/test_packagelayout.py]: Ugh, the copy-and-paste devil. Thanks.

**Comment by EnTeQuAk** [comment]
r+ once travis passes

**Comment by wagnerand** [comment]
Thank you!

**Pull request merged**